hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -511,8 +512,13 @@ protected HoodieTable doInitTable(HoodieTableMetaClient metaClient, Option<Strin
    */
   private void initializeMetadataTable(Option<String> inFlightInstantTimestamp) {
     if (config.isMetadataTableEnabled()) {
-      SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config,
+      HoodieTableMetadataWriter writer = SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config,
           context, Option.empty(), inFlightInstantTimestamp);
+      try {
+        writer.close();
```

Comment:
How about using try-with-resources?

Reply:
this has just 1 line. so not sure how we can get it w/ try w/ resources. we might need to have atleast 1 additional line to go into the try block. here we don't have any.